src/main/java/com/openlattice/authorization/EdmAuthorizationHelper.java

```diff
@@ -330,12 +329,52 @@ public EdmAuthorizationHelper( EdmManager edm, AuthorizationManager authz, Entit
             Set<UUID> entitySetIds,
             EnumSet<Permission> requiredPermissions,
             Set<Principal> principals ) {
-        return ( entitySetIds.isEmpty() )
-                ? Maps.newHashMap()
-                : entitySetIds.stream().collect( Collectors.toMap(
-                Function.identity(),
-                entitySetId -> getAuthorizedPropertyTypes( entitySetId, requiredPermissions, principals )
-        ) );
+        if ( entitySetIds.isEmpty() ) {
+            return Maps.newHashMap();
+        }
+
+        Map<UUID, Map<UUID, PropertyType>> entitySetIdsToPropertyTypes = entitySetManager
```

Comment:
I know there's a way for this to be less ugly and massive but the perf *should* be much better